core/src/main/java/org/springframework/ldap/odm/core/impl/ObjectMetaData.java

```diff
@@ -119,31 +125,34 @@ public ObjectMetaData(Class<?> clazz) {
         }
 
         // Get field meta-data - the @Attribute annotation
-        Field[] fields = clazz.getDeclaredFields();
-        for (Field field : fields) {
-            // So we can write to private fields
-            field.setAccessible(true);
-
-            // Skip synthetic or static fields
-            if (Modifier.isStatic(field.getModifiers()) || field.isSynthetic()) {
-                continue;
-            }
-
-            AttributeMetaData currentAttributeMetaData=new AttributeMetaData(field);
-            if (currentAttributeMetaData.isId()) {
-                if (idAttribute!=null) {
-                    // There can be only one id field
-                    throw new MetaDataException(
-                          String.format("You man have only one field with the %1$s annotation in class %2$s", Id.class, clazz));
+        ReflectionUtils.doWithFields(clazz, new ReflectionUtils.FieldCallback() {
```

Comment:
It's not clear to me the purpose of this change to use `doWithFields`. Is this change necessary to address ODM inheritance?

Reply:
If I recall correctly, clazz.getDeclaredFields(); will only return fields declaired directly in the class and not any inherited fields. the "doWithFields" method will recurse through the type inheritance tree to get all fields annotateed with spring ldap annotations.

https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getDeclaredFields--